packages/vercel/src/index.ts

```diff
@@ -0,0 +1,620 @@
+								dest: func,
+							});
+						}
+					} else {
+						const entryFile = new URL(_serverEntry, _buildTempFolder);
+						if (isr) {
+							const isrConfig = typeof isr === 'object' ? isr : {};
+							await builder.buildServerlessFolder(entryFile, NODE_PATH, _config.root);
+							if (isrConfig.exclude?.length) {
+								const dest = _middlewareEntryPoint ? MIDDLEWARE_PATH : NODE_PATH;
+								for (const route of isrConfig.exclude) {
+									// vercel interprets src as a regex pattern, so we need to escape it
+									routeDefinitions.push({ src: escapeRegex(route), dest });
+								}
+							}
+							await builder.buildISRFolder(entryFile, '_isr', isrConfig, _config.root);
+							for (const route of routes) {
+								const src = route.pattern.source;
+								const dest = src.startsWith('^\\/_image') ? NODE_PATH : ISR_PATH;
+								if (!route.prerender) routeDefinitions.push({ src, dest });
+							}
+						} else {
+							await builder.buildServerlessFolder(entryFile, NODE_PATH, _config.root);
+							const dest = _middlewareEntryPoint ? MIDDLEWARE_PATH : NODE_PATH;
+							for (const route of routes) {
+								if (!route.prerender) routeDefinitions.push({ src: route.pattern.source, dest });
+							}
+						}
+					}
+					if (_middlewareEntryPoint) {
+						await builder.buildMiddlewareFolder(
+							_middlewareEntryPoint,
+							MIDDLEWARE_PATH,
+							middlewareSecret
+						);
+					}
+				}
+				const fourOhFourRoute = routes.find((route) => route.pathname === '/404');
+				const destination =
+					buildOutput === 'server'
+						? new URL('./config.json', _config.outDir)
+						: new URL('./.vercel/output/config.json', _config.root);
+				const finalRoutes = [
+					...getRedirects(routes, _config),
+					{
+						src: `^/${_config.build.assets}/(.*)$`,
+						headers: { 'cache-control': 'public, max-age=31536000, immutable' },
+						continue: true,
+					},
+					{ handle: 'filesystem' },
+				];
+				if (buildOutput === 'server') {
+					finalRoutes.push(...routeDefinitions);
+				}
+
+				if (fourOhFourRoute) {
+					if (buildOutput === 'server') {
+						finalRoutes.push({
+							src: '/.*',
+							dest: fourOhFourRoute.prerender
+								? '/404.html'
+								: _middlewareEntryPoint
+									? MIDDLEWARE_PATH
+									: NODE_PATH,
+							status: 404,
+						});
+					} else {
+						finalRoutes.push({
+							src: '/.*',
+							dest: '/404.html',
+							status: 404,
+						});
+					}
+				}
+
+				let images: VercelImageConfig | undefined;
+				if (imageService || imagesConfig) {
+					if (imagesConfig) {
+						images = {
+							...imagesConfig,
+							domains: [...imagesConfig.domains, ..._config.image.domains],
+							remotePatterns: [
+								// ...(imagesConfig.remotePatterns ?? []),
+								// ..._config.image.remotePatterns,
+							],
+						};
+						if (imagesConfig.remotePatterns) {
+							images.remotePatterns?.push(...imagesConfig.remotePatterns);
+						}
```

Comment:
I think we can remove the comments here since they're unused. Or could combine both as:
```suggestion
							remotePatterns: [
								...(imagesConfig.remotePatterns ?? []),
							],
						};
```